internal/app/machined/acompat/acompat.go

```diff
@@ -0,0 +1,14 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package acompat provides compatibility with gRPC 1.67.0 and later.
+package acompat
+
+import "os"
+
+func init() {
+	if err := os.Setenv("GRPC_ENFORCE_ALPN_ENABLED", "false"); err != nil {
```

Comment:
[root.Run](https://github.com/siderolabs/talos/pull/9504/files#diff-aacec839c1945142f93d352109472284cde83752df4d6151f92f120cb7ac8fcbR13) which [runs](https://github.com/siderolabs/talos/pull/9504/files#diff-c2982427f2fcd2beba4ed9dd31207f613e2166dbb6cef670c11b7b3437f2fa49R300-R327) both `apid` and `trustd` among others. This should suffice no?

Reply:
ah, I was thinking the other way around, as we run them, we can actually pass the variable without hacks at all.

I think it would be way more clean:

`app/init` runs `app/machined`, so it can set any env (it already does)
`app/machined` runs both `apid` & `trustd` (don't need for trustd) as a service, so enable it explicitly instead of this init function side-effect which will be hard to track and clean up.

Reply:
Done [here](https://github.com/siderolabs/talos/pull/9504/files#diff-b7574acef7393c34b55edc407e3e1daf65f5dcd8945ef8c5b1ad86681f285c9fR168).